src/thunderbots/software/geom/util.cpp

```diff
@@ -665,18 +660,35 @@ std::vector<Point> lineIntersection(const Seg &a, const Seg &b)
                                   (a.end - a.start)};
 }
 
-// ported code
+// shamelessly copy-pasted from RoboJackets
 Vector lineIntersection(
     const Vector &a, const Vector &b, const Vector &c, const Vector &d)
 {
-    // TODO figure out why this is asserting
-    // assert(std::abs((d - c).cross(b - a)) > EPS);
-    if (std::fabs((d - c).cross(b - a)) < EPS)
+    Seg line1 (a, b), line2 (c, d);
+    double x1 = line1.start.x();
+    double y1 = line1.start.y();
+    double x2 = line1.end.x();
+    double y2 = line1.end.y();
+    double x3 = line2.start.x();
+    double y3 = line2.start.y();
+    double x4 = line2.end.x();
+    double y4 = line2.end.y();
+
+    double denom = (x1 - x2) * (y3 - y4) - (y1 - y2) * (x3 - x4);
+    if (denom == 0)
     {
-        // LOG_WARN(u8"Cross product problem again");
+        throw std::out_of_range("fuck");
```

Comment:
Although I would find running into this exception pretty funny, let's use something more descriptive and less profane.

Preferably, this is where it would be good to return an `std:optional<Point>`, since we are not "guaranteed" an intersection if the lines are parallel.

Reply:
Changed to use std::optional